src/app/lib/utilities/preprocessor/rules/injectAds.js

```diff
@@ -0,0 +1,56 @@
+import compose from 'ramda/src/compose';
+import {
+  getJsonContent,
+  getBlocks,
+  mergeJsonRawWithBlocks,
+} from './addIdsToBlocks';
+
+const adBlock = {
+  model: {
+    attributes: [],
+    text: '',
+  },
+  type: 'ad',
+};
+
+const mapInjectAdsWithinText = blocks => {
+  let paragraphCount = 0;
+  let adCount = 0;
+
+  return blocks.reduce((accBlocks, currentBlock) => {
+    const withoutAd = [...accBlocks, currentBlock];
+
+    if (currentBlock.type !== 'text') {
+      return withoutAd;
+    }
+
+    const paragraphs = getBlocks(currentBlock);
+    paragraphCount += paragraphs.length;
+
+    const withAd = () => {
+      adCount += 1;
+      return [...accBlocks, currentBlock, adBlock];
+    };
+
+    if (paragraphCount > 5 && adCount === 0) {
```

Comment:
These values are hardcoded for now but would be pulled out some form of config where the values can be different per service.